### ISSUE TITLE
HTTP 500 error when trying to access https://<my ip address>

### ISSUE THREAD

**Issue by glennb13** [nextcloud/nextcloudpi#482]
Hi, newbie here. I am not able to access the nextcloud via https://my-ip-address
webpage is showing HTTP 500 error
It is a fresh install via Etcher, did the pi password change, ssh, entered country locale, and enabled wifi.
it is a Rasberry Pi B+

https://my-ip-address:4443 and  https://my-ip-address:4443/wizard is showing properly though.

Please help point me in the right direction. and Thank you in advance 
output of `sudo ncp-report`
<--! Paste this in GitHub report -->

<details>
<summary>NextCloudPi diagnostics</summary>

```

NextCloudPi version  v0.53.27
NextCloudPi image    NextCloudPi_03-18-18
distribution         Raspbian GNU/Linux 9 \n \l
automount            no
USB devices          none
datadir              /var/www/nextcloud/data
data in SD           yes
data filesystem      ext2/ext3
data disk usage      1.9G/118G
rootfs usage         1.9G/118G
swapfile             /var/swap
Nextcloud check      error
HTTPD service        up
PHP service          up
MariaDB service      up
Redis service        up
Postfix service      up
internet check       no }
port check 80        open
port check 443       open
IP                   192.168.0.155
gateway              192.168.0.1
interface            wlan0
certificates         none
NAT loopback         no
uptime               11min

```
</details>

<details>
<summary>Nextcloud configuration</summary>

```

An unhandled exception has been thrown:
Doctrine\DBAL\DBALException: Failed to connect to the database: An exception occured in driver: SQLSTATE[HY000] [1045] Access denied for user 'ncadmin'@'localhost' (using password: YES) in /var/www/nextcloud/lib/private/DB/Connection.php:64
Stack trace:
#0 /var/www/nextcloud/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Connection.php(429): OC\DB\Connection->connect()
#1 /var/www/nextcloud/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Connection.php(389): Doctrine\DBAL\Connection->getDatabasePlatformVersion()
#2 /var/www/nextcloud/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Connection.php(328): Doctrine\DBAL\Connection->detectDatabasePlatform()
#3 /var/www/nextcloud/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Connection.php(623): Doctrine\DBAL\Connection->getDatabasePlatform()
#4 /var/www/nextcloud/lib/private/DB/Connection.php(151): Doctrine\DBAL\Connection->setTransactionIsolation(2)
#5 /var/www/nextcloud/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/DriverManager.php(172): OC\DB\Connection->__construct(Array, Object(Doctrine\DBAL\Driver\PDOMySql\Driver), Object(Doctrine\DBAL\Configuration), Object(Doctrine\Common\EventManager))
#6 /var/www/nextcloud/lib/private/DB/ConnectionFactory.php(152): Doctrine\DBAL\DriverManager::getConnection(Array, Object(Doctrine\DBAL\Configuration), Object(Doctrine\Common\EventManager))
#7 /var/www/nextcloud/lib/private/Server.php(628): OC\DB\ConnectionFactory->getConnection('mysql', Array)
#8 /var/www/nextcloud/3rdparty/pimple/pimple/src/Pimple/Container.php(113): OC\Server->OC\{closure}(Object(OC\Server))
#9 /var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php(116): Pimple\Container->offsetGet('OCP\\IDBConnecti...')
#10 /var/www/nextcloud/lib/private/ServerContainer.php(132): OC\AppFramework\Utility\SimpleContainer->query('OCP\\IDBConnecti...')
#11 /var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php(164): OC\ServerContainer->query('OCP\\IDBConnecti...')
#12 /var/www/nextcloud/3rdparty/pimple/pimple/src/Pimple/Container.php(109): OC\AppFramework\Utility\SimpleContainer->OC\AppFramework\Utility\{closure}(Object(OC\Server))
#13 /var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php(116): Pimple\Container->offsetGet('DatabaseConnect...')
#14 /var/www/nextcloud/lib/private/ServerContainer.php(132): OC\AppFramework\Utility\SimpleContainer->query('DatabaseConnect...')
#15 /var/www/nextcloud/lib/private/Server.php(1511): OC\ServerContainer->query('DatabaseConnect...')
#16 /var/www/nextcloud/lib/private/Server.php(332): OC\Server->getDatabaseConnection()
#17 /var/www/nextcloud/3rdparty/pimple/pimple/src/Pimple/Container.php(113): OC\Server->OC\{closure}(Object(OC\Server))
#18 /var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php(116): Pimple\Container->offsetGet('OC\\Authenticati...')
#19 /var/www/nextcloud/lib/private/ServerContainer.php(132): OC\AppFramework\Utility\SimpleContainer->query('OC\\Authenticati...')
#20 /var/www/nextcloud/lib/private/Server.php(336): OC\ServerContainer->query('OC\\Authenticati...')
#21 /var/www/nextcloud/3rdparty/pimple/pimple/src/Pimple/Container.php(113): OC\Server->OC\{closure}(Object(OC\Server))
#22 /var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php(116): Pimple\Container->offsetGet('OC\\Authenticati...')
#23 /var/www/nextcloud/lib/private/ServerContainer.php(132): OC\AppFramework\Utility\SimpleContainer->query('OC\\Authenticati...')
#24 /var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php(164): OC\ServerContainer->query('OC\\Authenticati...')
#25 /var/www/nextcloud/3rdparty/pimple/pimple/src/Pimple/Container.php(109): OC\AppFramework\Utility\SimpleContainer->OC\AppFramework\Utility\{closure}(Object(OC\Server))
#26 /var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php(116): Pimple\Container->offsetGet('OC\\Authenticati...')
#27 /var/www/nextcloud/lib/private/ServerContainer.php(132): OC\AppFramework\Utility\SimpleContainer->query('OC\\Authenticati...')
#28 /var/www/nextcloud/lib/private/Server.php(352): OC\ServerContainer->query('OC\\Authenticati...')
#29 /var/www/nextcloud/3rdparty/pimple/pimple/src/Pimple/Container.php(113): OC\Server->OC\{closure}(Object(OC\Server))
#30 /var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php(116): Pimple\Container->offsetGet('OCP\\IUserSessio...')
#31 /var/www/nextcloud/lib/private/ServerContainer.php(132): OC\AppFramework\Utility\SimpleContainer->query('OCP\\IUserSessio...')
#32 /var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php(164): OC\ServerContainer->query('OCP\\IUserSessio...')
#33 /var/www/nextcloud/3rdparty/pimple/pimple/src/Pimple/Container.php(109): OC\AppFramework\Utility\SimpleContainer->OC\AppFramework\Utility\{closure}(Object(OC\Server))
#34 /var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php(116): Pimple\Container->offsetGet('UserSession')
#35 /var/www/nextcloud/lib/private/ServerContainer.php(132): OC\AppFramework\Utility\SimpleContainer->query('UserSession')
#36 /var/www/nextcloud/lib/private/Server.php(1380): OC\ServerContainer->query('UserSession')
#37 /var/www/nextcloud/lib/private/Server.php(688): OC\Server->getUserSession()
#38 /var/www/nextcloud/3rdparty/pimple/pimple/src/Pimple/Container.php(113): OC\Server->OC\{closure}(Object(OC\Server))
#39 /var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php(116): Pimple\Container->offsetGet('OC\\App\\AppManag...')
#40 /var/www/nextcloud/lib/private/ServerContainer.php(132): OC\AppFramework\Utility\SimpleContainer->query('OC\\App\\AppManag...')
#41 /var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php(164): OC\ServerContainer->query('OC\\App\\AppManag...')
#42 /var/www/nextcloud/3rdparty/pimple/pimple/src/Pimple/Container.php(109): OC\AppFramework\Utility\SimpleContainer->OC\AppFramework\Utility\{closure}(Object(OC\Server))
#43 /var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php(116): Pimple\Container->offsetGet('AppManager')
#44 /var/www/nextcloud/lib/private/ServerContainer.php(132): OC\AppFramework\Utility\SimpleContainer->query('AppManager')
#45 /var/www/nextcloud/lib/private/Server.php(1684): OC\ServerContainer->query('AppManager')
#46 /var/www/nextcloud/lib/private/legacy/app.php(332): OC\Server->getAppManager()
#47 /var/www/nextcloud/lib/private/legacy/app.php(113): OC_App::getEnabledApps()
#48 /var/www/nextcloud/lib/base.php(661): OC_App::loadApps(Array)
#49 /var/www/nextcloud/lib/base.php(1080): OC::init()
#50 /var/www/nextcloud/console.php(46): require_once('/var/www/nextcl...')
#51 /var/www/nextcloud/occ(11): require_once('/var/www/nextcl...')
#52 {main}
```
</details>

<details>
<summary>HTTPd logs</summary>

```

[Mon Apr 16 12:54:39.169196 2018] [authz_host:error] [pid 860:tid 1934885936] [client 192.168.0.190:52905] AH01753: access check of 'localhost' to /img/ncp-logo.svg failed, reason: unable to get the remote host name, referer: https://192.168.0.155:4443/
[Mon Apr 16 12:54:39.169451 2018] [authz_host:error] [pid 860:tid 1918108720] [client 192.168.0.190:52905] AH01753: access check of 'localhost' to /img/loading-small.gif failed, reason: unable to get the remote host name, referer: https://192.168.0.155:4443/
[Mon Apr 16 12:54:39.169451 2018] [authz_host:error] [pid 860:tid 1926497328] [client 192.168.0.190:52905] AH01753: access check of 'localhost' to /img/poweroff.svg failed, reason: unable to get the remote host name
[Mon Apr 16 12:54:39.219436 2018] [authz_host:error] [pid 860:tid 1943274544] [client 192.168.0.190:52905] AH01753: access check of 'localhost' to /img/reboot.svg failed, reason: unable to get the remote host name
[Mon Apr 16 12:54:45.642479 2018] [authz_host:error] [pid 860:tid 1876165680] [client 192.168.0.190:52905] AH01753: access check of 'localhost' to /wizard/ failed, reason: unable to get the remote host name
[Mon Apr 16 12:54:45.788924 2018] [authz_host:error] [pid 860:tid 1876165680] [client 192.168.0.190:52905] AH01753: access check of 'localhost' to /wizard/ failed, reason: unable to get the remote host name
[Mon Apr 16 12:54:45.789339 2018] [authz_host:error] [pid 860:tid 1876165680] [client 192.168.0.190:52905] AH01753: access check of 'localhost' to /wizard/index.php failed, reason: unable to get the remote host name
[Mon Apr 16 12:54:45.888022 2018] [authz_host:error] [pid 860:tid 1876165680] [client 192.168.0.190:52905] AH01753: access check of 'localhost' to /wizard/index.php failed, reason: unable to get the remote host name
[Mon Apr 16 12:54:45.933588 2018] [authz_host:error] [pid 860:tid 1884554288] [client 192.168.0.190:52905] AH01753: access check of 'localhost' to /wizard/CSS/wizard.css failed, reason: unable to get the remote host name, referer: https://192.168.0.155:4443/wizard/
[Mon Apr 16 12:54:45.933592 2018] [authz_host:error] [pid 860:tid 1867777072] [client 192.168.0.190:52905] AH01753: access check of 'localhost' to /wizard/bootstrap/css/bootstrap.min.css failed, reason: unable to get the remote host name, referer: https://192.168.0.155:4443/wizard/
[Mon Apr 16 12:54:45.934056 2018] [authz_host:error] [pid 860:tid 1892942896] [client 192.168.0.190:52905] AH01753: access check of 'localhost' to /wizard/JS/jquery-latest.js failed, reason: unable to get the remote host name, referer: https://192.168.0.155:4443/wizard/
[Mon Apr 16 12:54:45.935358 2018] [authz_host:error] [pid 860:tid 1884554288] [client 192.168.0.190:52905] AH01753: access check of 'localhost' to /wizard/bootstrap/js/bootstrap.min.js failed, reason: unable to get the remote host name, referer: https://192.168.0.155:4443/wizard/
[Mon Apr 16 12:54:45.946589 2018] [authz_host:error] [pid 860:tid 1901331504] [client 192.168.0.190:52905] AH01753: access check of 'localhost' to /wizard/JS/jquery.bootstrap.wizard.js failed, reason: unable to get the remote host name, referer: https://192.168.0.155:4443/wizard/
[Mon Apr 16 12:54:45.948755 2018] [authz_host:error] [pid 860:tid 1850999856] [client 192.168.0.190:52905] AH01753: access check of 'localhost' to /wizard/img/ncp-logo.svg failed, reason: unable to get the remote host name, referer: https://192.168.0.155:4443/wizard/
[Mon Apr 16 12:54:45.948755 2018] [authz_host:error] [pid 860:tid 1859388464] [client 192.168.0.190:52905] AH01753: access check of 'localhost' to /wizard/JS/wizard.js failed, reason: unable to get the remote host name, referer: https://192.168.0.155:4443/wizard/
[Mon Apr 16 12:54:45.997015 2018] [authz_host:error] [pid 860:tid 1792279600] [client 192.168.0.190:52905] AH01753: access check of 'localhost' to /img/menu.svg failed, reason: unable to get the remote host name, referer: https://192.168.0.155:4443/wizard/CSS/wizard.css
[Mon Apr 16 12:54:46.012961 2018] [authz_host:error] [pid 860:tid 1968440368] [client 192.168.0.190:52905] AH01753: access check of 'localhost' to /wizard/img/nc-logo.png failed, reason: unable to get the remote host name, referer: https://192.168.0.155:4443/wizard/
[Mon Apr 16 12:54:46.241730 2018] [authz_host:error] [pid 860:tid 1817445424] [client 192.168.0.190:52905] AH01753: access check of 'localhost' to /ncp-output.php failed, reason: unable to get the remote host name, referer: https://192.168.0.155:4443/wizard/
[Mon Apr 16 12:54:46.242669 2018] [authz_host:error] [pid 860:tid 1783890992] [client 192.168.0.190:52905] AH01753: access check of 'localhost' to /img/favicon.png failed, reason: unable to get the remote host name, referer: https://192.168.0.155:4443/wizard/
[Mon Apr 16 12:54:50.567935 2018] [proxy_fcgi:error] [pid 860:tid 1817445424] [client 192.168.0.190:52905] AH01071: Got error 'PHP message: PHP Warning:  touch(): Utime failed: Permission denied in /var/www/ncp-web/ncp-output.php on line 81\n', referer: https://192.168.0.155:4443/wizard/

```
</details>

<details>
<summary>Database logs</summary>

```

2018-03-18 23:12:31 1639838528 [Note] /usr/sbin/mysqld: Shutdown complete

2018-04-16 12:49:00 1988702208 [Note] InnoDB: Using mutexes to ref count buffer pool pages
2018-04-16 12:49:00 1988702208 [Note] InnoDB: The InnoDB memory heap is disabled
2018-04-16 12:49:00 1988702208 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
2018-04-16 12:49:00 1988702208 [Note] InnoDB: GCC builtin __atomic_thread_fence() is used for memory barrier
2018-04-16 12:49:00 1988702208 [Note] InnoDB: Compressed tables use zlib 1.2.8
2018-04-16 12:49:00 1988702208 [Note] InnoDB: Using Linux native AIO
2018-04-16 12:49:00 1988702208 [Note] InnoDB: Using generic crc32 instructions
2018-04-16 12:49:00 1988702208 [Note] InnoDB: Initializing buffer pool, size = 128.0M
2018-04-16 12:49:00 1988702208 [Note] InnoDB: Completed initialization of buffer pool
2018-04-16 12:49:01 1988702208 [Note] InnoDB: Highest supported file format is Barracuda.
2018-04-16 12:49:01 1988702208 [Note] InnoDB: 128 rollback segment(s) are active.
2018-04-16 12:49:01 1988702208 [Note] InnoDB: Waiting for purge to start
2018-04-16 12:49:01 1988702208 [Note] InnoDB:  Percona XtraDB (http://www.percona.com) 5.6.35-80.0 started; log sequence number 3394410
2018-04-16 12:49:01 1447031616 [Note] InnoDB: Dumping buffer pool(s) not yet started
2018-04-16 12:49:01 1988702208 [Note] Plugin 'FEEDBACK' is disabled.
2018-04-16 12:49:01 1988702208 [Note] Server socket created on IP: '127.0.0.1'.
2018-04-16 12:49:01 1988702208 [Note] /usr/sbin/mysqld: ready for connections.
Version: '10.1.23-MariaDB-9+deb9u1'  socket: '/var/run/mysqld/mysqld.sock'  port: 3306  Raspbian 9.0

```
</details>

<details>
<summary>Nextcloud logs</summary>

```

{"reqId":"D49yStIUofSIu5BPw4JN","level":4,"time":"2018-03-18T20:30:23+00:00","remoteAddr":"","user":"--","app":"cron","method":"--","url":"--","message":"Not installed","userAgent":"--","version":""}
{"reqId":"6hum7Mc00NWMhxp9PcrW","level":4,"time":"2018-03-18T20:45:16+00:00","remoteAddr":"","user":"--","app":"cron","method":"--","url":"--","message":"Not installed","userAgent":"--","version":""}
{"reqId":"0pGslTZa49CMdLXvcchQ","level":4,"time":"2018-03-18T21:00:15+00:00","remoteAddr":"","user":"--","app":"cron","method":"--","url":"--","message":"Not installed","userAgent":"--","version":""}
{"reqId":"QjDmMDD3iahD8pMaAFP8","level":4,"time":"2018-03-18T21:15:13+00:00","remoteAddr":"","user":"--","app":"cron","method":"--","url":"--","message":"Not installed","userAgent":"--","version":""}
{"reqId":"71jTTe2yIaOKIUUbutpc","level":4,"time":"2018-03-18T21:30:17+00:00","remoteAddr":"","user":"--","app":"cron","method":"--","url":"--","message":"Not installed","userAgent":"--","version":""}
{"reqId":"NsMw6FKjigDIfYUwYKCY","level":4,"time":"2018-03-18T21:45:14+00:00","remoteAddr":"","user":"--","app":"cron","method":"--","url":"--","message":"Not installed","userAgent":"--","version":""}
{"reqId":"Bx8ujWZN1DHx3NGOxtaI","level":4,"time":"2018-03-18T22:00:15+00:00","remoteAddr":"","user":"--","app":"cron","method":"--","url":"--","message":"Not installed","userAgent":"--","version":""}
{"reqId":"Kye7bXpvbCdrgp6nTa6v","level":4,"time":"2018-03-18T22:15:14+00:00","remoteAddr":"","user":"--","app":"cron","method":"--","url":"--","message":"Not installed","userAgent":"--","version":""}
{"reqId":"9RHG3Kjh6F4dFTUJevgp","level":4,"time":"2018-03-18T22:30:22+00:00","remoteAddr":"","user":"--","app":"cron","method":"--","url":"--","message":"Not installed","userAgent":"--","version":""}
{"reqId":"fNP4SWER3cq6MoGyG4lg","level":4,"time":"2018-03-18T22:45:25+00:00","remoteAddr":"","user":"--","app":"cron","method":"--","url":"--","message":"Not installed","userAgent":"--","version":"13.0.1.1"}

```
</details>

**Comment by glennb13** [comment]
Please help point me to the right direction

**Comment by chiyi4488** [comment]
Can you use HTTPS?

**Comment by glennb13** [comment]
that is actually what is weird. I am using https:// but it gives me;
HTTP 500 error
That’s odd... the website can’t display this page
That's in I.E.  In Opera it gives me even less informative error of;
This page isn’t working
192.168.0.155 is currently unable to handle this request.

If I try http:// it shows the standard self signed certificate warning, after accepting to continue with self signed cert it gives me the same error.

**Comment by nachoparker** [comment]
It's weird that only you are seeing this. I would look for [hardware issues](https://github.com/nextcloud/nextcloudpi/wiki/FAQ#why-is-nextcloud-so-slow--failing-randomly--getting-server-errors-even-with-a-fresh-image)

**Comment by chiyi4488** [comment]
When I reboot, I also showed 500.

**Comment by glennb13** [comment]
Thank you nachoparker!

That was an expensive problem.  I bought an new 128gb SD card, formatted to ext4 with minitool partition, and flashed with etcher, did the country locale, change passwords and enabled wifi. And this time it worked!  I'll reformat my SD hopefully it's not totally broken.

P.S. I ran into that "trusted domain" issue asking to add my IP to the config.php from /var/www/nextcloud/config/config.php and I saw the config.php has gotten a different IP;
  array (
    0 => 'localhost',
    5 => 'nextcloudpi.local',
    1 => '192.168.0.132',
Not sure if this was hardcoded, anyways I changed it to my IP and everything got fixed.

Thanks again
